tools/ci/run_tests.sh

```diff
@@ -17,12 +17,16 @@ echo "backend : agg" > matplotlibrc
 if [ "$COVERAGE" == "1" ] || [ "$COVERAGE" == true ]; then
     cp ../.coveragerc .;
     cp ../.codecov.yml .;
+    chmod -R a-w .
     # Run the tests and check for test coverage.
     coverage run -m pytest -c pyproject.toml -svv --doctest-modules --verbose --durations=10 --pyargs dipy
```

Comment:
- Here, since coverage is `True`, a `.coverage` is created and send to Codecov.  Unfortunately, it is impossible now.  So we might need to create the folder `for_testing` one level up instead of one level down.

- I would add `-o cache_dir=../..` or something similar to avoid the pytest warning below since we treat warnings as errors.

```
../venv/lib/python3.10/site-packages/_pytest/cacheprovider.py:445
  /home/runner/work/dipy/dipy/venv/lib/python3.10/site-packages/_pytest/cacheprovider.py:445: PytestCacheWarning: could not create cache path /home/runner/work/dipy/dipy/for_testing/.pytest_cache/v/cache/nodeids: [Errno 13] Permission denied: '/home/runner/work/dipy/dipy/for_testing/.pytest_cache'
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../venv/lib/python3.10/site-packages/_pytest/cacheprovider.py:397
  /home/runner/work/dipy/dipy/venv/lib/python3.10/site-packages/_pytest/cacheprovider.py:397: PytestCacheWarning: could not create cache path /home/runner/work/dipy/dipy/for_testing/.pytest_cache/v/cache/lastfailed: [Errno 13] Permission denied: '/home/runner/work/dipy/dipy/for_testing/.pytest_cache'
    config.cache.set("cache/lastfailed", self.lastfailed)

../venv/lib/python3.10/site-packages/_pytest/stepwise.py:56
  /home/runner/work/dipy/dipy/venv/lib/python3.10/site-packages/_pytest/stepwise.py:56: PytestCacheWarning: could not create cache path /home/runner/work/dipy/dipy/for_testing/.pytest_cache/v/cache/stepwise: [Errno 13] Permission denied: '/home/runner/work/dipy/dipy/for_testing/.pytest_cache'
    session.config.cache.set(STEPWISE_CACHE_DIR, [])
```